tests/unit-tests/frontend_wayland/test_screencopy_v1_damage_tracker.cpp

```diff
@@ -0,0 +1,465 @@
+    tracker.capture_on_damage(&frame_a);
+    executor.execute();
+
+    EXPECT_CALL(frame_a, capture(_));
+    StrictMock<MockFrame> frame_b{area};
+    tracker.capture_on_damage(&frame_b);
+    executor.execute();
+}
+
+TEST_F(DamageTrackerV1, allows_multiple_pending_frames_with_the_same_area_if_outputs_are_different)
+{
+    geom::Rectangle const area{{}, {20, 30}};
+    capture_frame_with_area(area);
+    wl_resource* output_a{reinterpret_cast<wl_resource*>(1)};
+    wl_resource* output_b{reinterpret_cast<wl_resource*>(2)};
+
+    {
+        NiceMock<MockFrame> frame_a{area};
+        frame_a.params.output = output_a;
+        tracker.capture_on_damage(&frame_a);
+
+        NiceMock<MockFrame> frame_b{area};
+        frame_b.params.output = output_b;
+        tracker.capture_on_damage(&frame_b);
+
+        executor.execute();
+    }
+
+    StrictMock<MockFrame> frame_a{area};
+    frame_a.params.output = output_a;
+    tracker.capture_on_damage(&frame_a);
+    executor.execute();
+
+    StrictMock<MockFrame> frame_b{area};
+    frame_b.params.output = output_b;
+    tracker.capture_on_damage(&frame_b);
+    executor.execute();
+
+    EXPECT_CALL(frame_a, capture(_));
+    EXPECT_CALL(frame_b, capture(_));
+    damage_whole_scene();
+    executor.execute();
+}
+
+TEST_F(DamageTrackerV1, captures_second_frame_on_second_area_immediately_when_second_area_is_already_damaged_and_first_area_has_pending_frame)
+{
+    geom::Rectangle const area_a{{ 0, 0}, {20, 30}};
+    geom::Rectangle const area_b{{25, 0}, {20, 30}};
+    capture_frame_with_area(area_a);
+    capture_frame_with_area(area_b);
+
+    damage_area(area_b);
+    executor.execute();
+
+    StrictMock<MockFrame> frame_a{area_a};
+    EXPECT_CALL(frame_a, capture(_)).Times(0);
+    tracker.capture_on_damage(&frame_a);
+    executor.execute();
+
+    StrictMock<MockFrame> frame_b{area_b};
+    EXPECT_CALL(frame_b, capture(_));
+    tracker.capture_on_damage(&frame_b);
+    executor.execute();
+}
+
+TEST_F(DamageTrackerV1, captures_with_full_damage_when_partial_damage_is_followed_by_full_damage)
+{
+    geom::Rectangle const area{{}, {20, 30}};
+    capture_frame_with_area(area);
+
+    damage_area(area);
+    executor.execute();
+
+    damage_whole_scene();
+    executor.execute();
+
+    StrictMock<MockFrame> frame{area};
+    EXPECT_CALL(frame, capture(Eq(std::nullopt)));
+    tracker.capture_on_damage(&frame);
+    executor.execute();
+}
+
+TEST_F(DamageTrackerV1, captures_with_full_damage_when_full_damage_is_followed_by_partial_damage)
+{
+    geom::Rectangle const area{{}, {20, 30}};
+    capture_frame_with_area(area);
+
+    damage_whole_scene();
+    executor.execute();
+
+    damage_area(area);
+    executor.execute();
+
+    StrictMock<MockFrame> frame{area};
+    EXPECT_CALL(frame, capture(Eq(std::nullopt)));
+    tracker.capture_on_damage(&frame);
+    executor.execute();
+}
+
+TEST_F(DamageTrackerV1, captured_with_combined_area_when_multiple_damages_happen_before_capture)
```

Comment:
The test names seem a bit strange.

`when_multiple_damages_happen_before_capture_then_captures_combined_damage_areas`?